# This file is part of HiGFXback

# requires
REQUIRES="autotools-wrappers-lt-build"

pkg-config --exists --print-errors $REQUIRES || exit 1

pkg-config --exists xz-build && REQUIRES="$REQUIRES xz-build"

# configure (ac-2.71; am-1.16; lt-2.4.6)
libtoolize -c -f; aclocal; autoheader; autoconf; automake -a -c
./configure --prefix=

# build
make

# install
install -d $DESTDIR/bin
install src/.libs/file $DESTDIR/bin
install -d $DESTDIR/include
install -m 644 src/magic.h $DESTDIR/include
install -d $DESTDIR/lib
install src/.libs/libmagic.so.1.* $DESTDIR/lib/libmagic.so.1
ln -sf libmagic.so.1 $DESTDIR/lib/libmagic.so
install -d $DESTDIR/lib/pkgconfig
install -m 644 libmagic.pc $DESTDIR/lib/pkgconfig
install -d $DESTDIR/share/misc
install -m 644 magic/magic.mgc $DESTDIR/share/misc

# build.pc
install -d $DESTDIR/share/pkgconfig
cat > $DESTDIR/share/pkgconfig/file-build.pc << EOF
Name: file
Version: 5.45
Description: File type identification utility and magic number recognition library
Requires: $REQUIRES

devel=\\
/include/magic.h \\
/lib/libmagic.so \\
/lib/pkgconfig/libmagic.pc

exec=\\
/bin/file \\
/lib/libmagic.so.1 \\
/share/misc/magic.mgc
EOF

exit

# patch
--- file-5.45.orig/configure.ac
+++ file-5.45/configure.ac
@@ -3,6 +3,8 @@
 AM_INIT_AUTOMAKE([subdir-objects foreign])
 m4_ifdef([AM_SILENT_RULES], [AM_SILENT_RULES([yes])])
 
+AM_MAINTAINER_MODE
+
 AC_CONFIG_HEADERS([config.h])
 AC_CONFIG_MACRO_DIR([m4])
 
# source
https://astron.com/pub/file/file-5.45.tar.gz
